src/lua/LuaBody.cpp

```diff
@@ -118,6 +118,55 @@ static int l_body_attr_path(lua_State *l)
 	return 1;
 }
 
+static int l_body_get_component(lua_State *l)
+{
+	Body *b = LuaObject<Body>::CheckFromLua(1);
+	std::string componentName = luaL_checkstring(l, 2);
+
+	BodyComponentDB::PoolBase *pool = BodyComponentDB::GetComponentType(componentName);
+	if (pool && pool->luaInterface) {
+		pool->luaInterface->PushToLua(b);
+		return 1;
+	}
+
+	if (componentName == "__properties") {
+		lua_pushnil(l);
+		return 1;
+	}
+
+	lua_getuservalue(l, 1);
+	lua_pushvalue(l, 2);
+	lua_rawget(l, -2);
+
+	return 1;
+}
+
+static int l_body_set_component(lua_State *l)
+{
+	Body *b = LuaObject<Body>::CheckFromLua(1);
```

Comment:
Unused variable.

Reply:
"Not a bug" - the CheckFromLua needs to validate the first parameter is the right userdata. This is a common pattern in LuaObject bindings.

Reply:
It seems that this is the first unused variable in Pioneer code that I and my static analyzer see, an he is unhappy. I have looked at all occurrences of `CheckFromLua`, the variable is always used afterwards. You don't have to put the result in a variable if you don't need it later, and you don't have to suffer a compile-time warning.

Reply:
Well, if it's for the sake of your static analyzer I suppose I can just discard the return value. :smile: